modules/core/shared/src/main/scala/gem/GuideGroup.scala

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+
+import cats._
+import cats.implicits._
+import cats.data.NonEmptyList
+import gem.enum.Guider
+
+/**
+ * Associates one or more guide probes or guide windows with guide stars.
+ */
+sealed trait GuideGroup extends Product with Serializable {
+
+  def guideStars: NonEmptyList[(Guider, Target)]
+
+}
+
+object GuideGroup {
+
+  /** Group identifier. */
+  final case class Id(toInt: Int) extends AnyVal
+
+  object Id {
+    /** Ids ordered by wrapped integer value. */
+    implicit val IdOrder: Order[Id] =
+      Order.by(_.toInt)
+  }
+
+  /**
+   * Guide "group" supertype for the majority of cases where there is a single
+   * guide star.
+   */
+  sealed trait SingleGuider extends GuideGroup {
```

Comment:
I kind of wonder if it would be worth it here to have `SingleGuider[G <: Guider]` and `def guider: G` and `Flamingos2.OI[Guider.F2OI.type]` so at least in the single-target case we would have that association statically. But I don't know if this will end up being useful or not.